### DISCUSSION TITLE
[WASM] How do I implement and configure a uno wasm app with MSAL authentication on Azure?

### DISCUSSION THREAD

**Discussion by tobizk04** [unoplatform/uno#16281] [Q&A]
It shoud be a app (wasm) that is hosted in azure and uses an asp.net REST service in the backend.
The access to the app should only allowed for AD users.
So we use the normal Web App Authentication.

I tried it with the following tutorial:
[https://platform.uno/docs/articles/interop/MSAL.html](https://platform.uno/docs/articles/interop/MSAL.html)

The following requirement:
- When opening the app in the browser, the user should first be redirected to normal Microsoft authentication (company tenant)
- The user logs in
- then the uno wasm should be loaded
-> and here is the problem
        the login is successful and the index.html is loaded
        but the loading of all files of the web assembly fails with the following errors:
        ->

Access to fetch at '... from origin 'https://test.azurewebsites.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
Access to fetch at '...ed/E3_CloudViewDemo.Wasm.clr') from origin 'https://test.azurewebsites.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
Access to fetch at '... from origin 'https://test.azurewebsites.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
Access to fetch at '...managed/Uno.UI.Runtime.WebAssembly.clr') from origin 'https://test.azurewebsites.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
Access to fetch at '...Foundation.clr') from origin 'https://test.azurewebsites.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
...
...
...

I think the main problem is that a don´t have the token from the authentication
How can I get this token to load the webassembly?

Does anyone have any idea what is going wrong?

**Comment by jeromelaban** [comment]
Thanks for the question. It seems that one of your sites is not having the right CORS headers set for the API calls to be working properly, make sure to add them at the origin: https://platform.uno/docs/articles/common-issues-wasm.html#webassembly-access-to-fetch-at-httpsxxxx-from-origin-httpxxxx-has-been-blocked-by-cors-policy.
